core/network/src/protocol/sync.rs

```diff
@@ -711,16 +757,18 @@ impl<B: BlockT> ChainSync<B> {
 		}
 	}
 
-	/// Sets the new head of chain.
+	/// Signal that `best_header` has been queued for import and update the `ChainSync` state with
+	/// that information.
 	pub(crate) fn update_chain_info(&mut self, best_header: &B::Header) {
 		let hash = best_header.hash();
 		self.block_queued(&hash, best_header.number().clone())
 	}
 
 	/// Call when a node announces a new block.
 	///
-	/// If true is returned, then the caller MUST try to import passed header (call `on_block_data).
+	/// If true is returned, then the caller MUST try to import passed header (call `on_block_data`).
 	/// The network request isn't sent in this case.
+	/// FH: Why is both hash and header passed? Hash is in the header?
```

Comment:
I see no reason both of these should be passed in?

Reply:
Agree, seems redundant.

Reply:
A bit of performance optimization to avoid rehashing the header again.